lib/common-lib.sh

```diff
@@ -5,28 +5,27 @@
 
 IFS=$'\n'
 
+# If the repo contains a .formatting-directory file, only files in the specified directories will be returned (one directory per line).
+# If .formatting-directory doesn't exist then all files in the repository are checked.
+function directories_to_check() {
+    locations_to_diff=''
+	[ -e ".formatting-directory" ] && locations_to_diff=$( cat .formatting-directory | sed 's/ /\\ /g' );
+}
+
 # Returns a list of Objective-C files to format.
 # No parameters: return staged ObjC files only
 # Optional parameter: a git SHA. Returns a list of all ObjC files which have changed since that SHA
```

Comment:
It still might be helpful to add a note about how in some cases, we won't get *all* files that have changed.